components/Buttons/MarkPlayedButton.vue

```diff
@@ -0,0 +1,70 @@
+<template>
+  <v-btn v-if="canMarkWatched" icon dark @click.stop.prevent="togglePlayed">
+    <v-icon :color="isPlayed ? 'primary' : 'white'"> mdi-check </v-icon>
+  </v-btn>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { mapActions } from 'vuex';
+import { BaseItemDto } from '@jellyfin/client-axios';
+
+export default Vue.extend({
+  props: {
+    item: {
+      type: Object as () => BaseItemDto,
+      required: true
+    }
+  },
+  data() {
+    return {
+      isPlayed: false
+    };
+  },
+  computed: {
+    canMarkWatched(): boolean {
+      if (
+        ['Series', 'Season', 'BoxSet', 'AudioPodcast', 'AudioBook'].includes(
+          this.item.Type || ''
+        )
+      ) {
+        return true;
+      }
+
+      if (this.item.MediaType === 'Video' && this.item.Type !== 'TvChannel') {
+        return true;
+      }
+
+      return false;
```

Comment:
We have canPlay in a mixin for this. You can mix it with the check of ``item.UserData.Played`` existence if you want better "safety".

Reply:
The sets of canPlay and canBeLiked are slightly different

canLike
Type: Series, Season, BoxSet, AudioPodcast, AudioBook
MediaType: Video (Type !== TvChannel)

canPlay
Type: MusicGenre Season Series, BoxSet, MusicAlbum, MusicArtist, Playlist
MediaType: Video Audio

Reply:
I would place it in the mixin anyway, so it can be revisited later in case it's needed easily. Right now we have these kind of things scattered all around, which makes maintenance more difficult.

Reply:
I was thinking about this earlier, I'm probably going to try and collate all those kind of tools to one single file, I'll probably move it out the mixin to `itemHelper` since it doesn't really need to be a mixin

Reply:
Yeah, that's what I meant, no need for a new mixin, just add it to itemHelper.